lib/turn.rb

```diff
@@ -0,0 +1,80 @@
+class Turn
+  attr_reader :player1, :player2, :spoils_of_war
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+  end
+
+  def type
+    if player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) && player1.deck.rank_of_card_at(2) == player2.deck.rank_of_card_at(2)
+      :mutually_assured_distruction
+    elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) &&  player1.deck.rank_of_card_at(2) != player2.deck.rank_of_card_at(2)
+      :war
+    else @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
+      :basic
+    end
+  end
+
+  def winner
+  if type == :mutually_assured_distruction
+    "No Winner"
+  elsif type == :war
+    war_winner
```

Comment:
I really love that you broke this logic out to its own method. Very smart programming. This makes the code so much more readable and easy to change and test. Same for the `basic_winner` method. If you here me talk about helper methods in class, this is exactly the type of thing I'm talking about. Great work 🔥